msal/src/Microsoft.Identity.Client/UIParent.cs

```diff
@@ -38,37 +39,35 @@ namespace Microsoft.Identity.Client
     /// </summary>
     public sealed class UIParent
     {
+        internal CoreUIParent CoreUIParent { get; private set; }
```

Comment:
>internal CoreUIParent CoreUIParent { get; private set; } [](start = 8, length = 56)

not sure I understand why this is a component of the UIParent vs. a base class. The name indicates otherwise to me. Can you pls. educate me on the background for this choice?

Reply:
I needed a class to hold the properties from ADAL and MSAL. I will need to move attributes of platform parameters from ADAL into CoreUIParent, so CoreUIParent will contain attributes of both (platform parameters and UIParent properties). 
If the word "Core" is weird, I can change it to something else.

Reply:
I'm thinking the xml comments of the class needs to explain the intention of the class. that will help me understand if it is more appropriate to name any of the classes differently. I'm not 100% sure what the intention of each of these are. Is the intention of CoreUIParent only to be a container for data/settings (properties)? Is there a reason not to put everything into Core? ... 

Aside: I think we talked about not using Core in our naming? Core and Base are generally names used to refer to certain inheritance hierarchies - which is probably what confused me the most in this case.

---
In reply to: [182136079](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1038#discussion_r182136079) [](ancestors = 182136079)

Reply:
I suppose I would consider the CoreUIParent the data/context object and the UIParent a factory class. Then the CoreUIParent (which could then be called UIParent) and the current UIParent could be called UIParentFactory ...

Any reason this is not all in the Core logic?

---
In reply to: [182141294](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1038#discussion_r182141294) [](ancestors = 182141294,182139799)

Reply:
It's not in core logic because UIParent is public and needs to remain in MSAL, and all the core classes are internal. CoreUIParent will hopefully be a single object representing properties of platform parameters and UIParent.